### PR TITLE
Added extra test to quantile extender to check consistency of db

### PR DESCRIPTION
**Pull request by Rlamboll**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Added feature which does something
- (`#XX <https://github.com/GranthamImperial/silicone/pull/XX>`_) Fixed bug identified in (`#YY <https://github.com/GranthamImperial/silicone/issues/YY>`_)
```

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -10,6 +10,7 @@ Added
 
 Changed
 ~~~~~~~
+- (`#149 <https://github.com/GranthamImperial/silicone/pull/149>`_) Sped up RMS closest
```
Comment (znicholls) [inline CHANGELOG.rst]: Needs an update right?

File: src/silicone/time_projectors/extend_latest_time_quantile.py [hunk 2]
```diff
@@ -157,6 +157,15 @@ def get_values_in_key_timepoint(idf):
                 quantiles = [0.5 if np.isnan(q) else q for q in quantiles]
             output_ts = target_df.timeseries()
             iamdf_ts = iamdf.timeseries()
+            if (
```
Comment (znicholls) [inline src/silicone/time_projectors/extend_latest_time_quantile.py]: Maybe refactor to something like

```
later_times_nans = (
iamdf_ts.loc[:, later_times]
                .dropna(axis=0, how="all")
                .isna()
                .sum()
                .sum()
)
if later_times_nans != 0:
```

Just so it's easier to understand what the intent is

Comment (znicholls) [reply src/silicone/time_projectors/extend_latest_time_quantile.py]: On that note, why do we `dropna(axis=0, how="all")` first? To drop out any timeseries which are entirely nan (and presumably get dealt with elsewhere)?

**Review by znicholls — COMMENTED** [review]
Nice, would have been a fun bug to find :)

File: src/silicone/time_projectors/extend_latest_time_quantile.py (hunk 2 above)
Comment (Rlamboll) [reply src/silicone/time_projectors/extend_latest_time_quantile.py]: Yes, it's quite common for some variables to never report at certain times and this is generally fine (and if not, will be caught  elsewhere). Thanks!

**Review by znicholls — APPROVED** [review]
lgtm

**Pull request merged**